test/e2e/README.md

```diff
@@ -1,37 +1,207 @@
+
+This comprehensive test suite ensures the OCM Agent service functions correctly across different deployment modes and scenarios, providing confidence in both basic operations and advanced fleet management capabilities.
+
+## Test Structure Improvements
+
+### Recent Refactoring
+
+The test suite has been significantly improved with the following changes:
+
+- **Common Test Utilities**: Shared test functions and structs have been moved to `pkg/consts/test/test.go` for better reusability
+- **Standardized Test Descriptions**: All `ginkgo.By` descriptions now follow a uniform format:
+  - `Setup:` for initialization and configuration steps
+  - `Step X:` for test execution steps  
+  - `Cleanup:` for teardown operations
+- **Improved Readability**: Test files are now more readable with consistent formatting and better organization
+
+## Prerequisites
+
+Before running the E2E tests, ensure you have:
+
+- Go 1.19+ installed
+- Access to a Kubernetes cluster with OCM Agent deployed
+- OCM credentials (token or client ID/secret)
+- `ginkgo` test runner installed
+
+## Installation
+
+Install the required dependencies:
 
-First, find the name of the `ocm-agent` pod:
 ```bash
-oc get pods -n openshift-ocm-agent-operator -l app=ocm-agent
+# Install Ginkgo test runner
+go install github.com/onsi/ginkgo/v2/ginkgo@latest
+
+# Build the E2E test binary
+make e2e-binary-build
 ```
 
-Use the pod name to set up port forwarding:
+## Running Tests Locally
+
+### Basic Setup
+
+1. **Build the test binary**:
+   ```bash
+   make e2e-binary-build
+   ```
+
+2. **Get cluster credentials**:
+   ```bash
+   ocm get /api/clusters_mgmt/v1/clusters/$INTERNAL_CLUSTERID/credentials | jq -r .kubeconfig > /path/to/kubeconfig
+   ```
+
+3. **Run the complete test suite**:
+   ```bash
+   DISABLE_JUNIT_REPORT=true KUBECONFIG=/path/to/kubeconfig ./bin/ginkgo --tags=osde2e -v test/e2e
+   ```
+
+### Running Specific Test Categories
+
+#### Common Tests (Basic Deployment)
 ```bash
-oc -n openshift-ocm-agent-operator port-forward <ocm-agent-pod-name> 8081:8081
+DISABLE_JUNIT_REPORT=true KUBECONFIG=/path/to/kubeconfig ginkgo run --tags=osde2e -vv --label-filter="OcmAgentCommon" test/e2e/
 ```
 
-Run the tests with the `OCM_AGENT_URL` environment variable set:
+#### Classic Mode Tests
 ```bash
-export OCM_TOKEN=$(ocm token)
-OCM_AGENT_URL=http://localhost:8081 DISABLE_JUNIT_REPORT=true KUBECONFIG=/(path-to)/kubeconfig ./bin/ginkgo --tags=osde2e -v test/e2e
+DISABLE_JUNIT_REPORT=true KUBECONFIG=/path/to/kubeconfig ginkgo run --tags=osde2e -vv --label-filter="OcmAgentClassic" test/e2e/
 ```
 
+#### Fleet Mode Tests
+```bash
+DISABLE_JUNIT_REPORT=true KUBECONFIG=/path/to/kubeconfig ginkgo run --tags=osde2e -vv --label-filter="OcmAgentHCP" test/e2e/
+```
+
+### Running with Port Forwarding
+
+When testing against a remote cluster, use port forwarding to access the OCM Agent service:
+
+1. **In one terminal session use the existing `build-and-run.sh` script:**
```

Comment:
This is not using port-forward from service hosted in the cluster, do we need to change this to port-foward the pod like below 
`oc -n openshift-ocm-agent-operator port-forward <ocm-agent-pod-name> 8081:8081`

Reply:
This step is mostly having a local run of ocm-agent tested against a cluster before we raise a PR and merge. For the dev and new build of ocm-agent it would be the same thing of having a local port or port-forwarding of the dev build. For in-cluster tests we would rely on E2E tests. Leveraging the script would make it easier to configure ocm-agent modes etc.